package NYT.NTableClient.NProto;

import "yt/ytlib/chunk_holder/chunk.proto";
import "yt/ytlib/table_client/schema.proto";

////////////////////////////////////////////////////////////////////////////////

message TBlockInfo
{
    required int32 block_index = 1;
    required int32 row_count = 2;
}

message TChunkChannel
{
    required TChannel channel = 1;
    repeated TBlockInfo blocks = 2;
}

message TKey
{
    repeated bytes values = 1;
}

message TKeySample
{
    required TKey key = 1;
    required int64 row_index = 2;
}

message TTableChunkAttributes
{
    extend NYT.NChunkHolder.NProto.TChunkAttributes
    {
        required TTableChunkAttributes table_attributes = 101;
    }

    repeated TChunkChannel chunk_channels = 1;

    // Id of the codec using for compressing blocks.
    required int32 codec_id = 2;

    // Number of table rows in the chunk.
    required int64 row_count = 3;

    // Uncompressed data size.
    // This has no exact meaning and depends on the physical layout of the data.
    required int64 uncompressed_size = 4;

    repeated bytes key_columns = 5;
    repeated TKeySample key_samples = 6;

    required bool sorted = 7;
}

////////////////////////////////////////////////////////////////////////////////

